File: backend/TenderGo.Models/Requests/RateUserRequest.cs
using System;
using System.Collections.Generic;
using System.ComponentModel.DataAnnotations;
using System.Linq;
using System.Text;
using System.Threading.Tasks;

namespace TenderGo.Models.Requests
{
    public class RateUserRequest
    {
        [Required]
        public string RatedUserId { get; set; } = null!;
        public int TenderId { get; set; }
        [Range(1, 5)]

        public int Score { get; set; }
        public string? Comment { get; set; }
      
    }
}

File: UI/tendergo_mobile/lib/shared/models/requests/rate_user_request.dart

class RateUserRequest {
  final String ratedByUserId;
  final String ratedUserId;
  final int tenderId;
  final int score;
  final String? comment;

  const RateUserRequest({
    required this.ratedByUserId,
    required this.ratedUserId,
    required this.tenderId,
    required this.score,
    this.comment,
  });

  Map<String, dynamic> toJson() => {
        'ratedByUserId': ratedByUserId,
        'ratedUserId': ratedUserId,
        'tenderId': tenderId,
        'score': score,
        'comment': comment,
       
      };
}